src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetStreamReferenceJsonConverter.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.JSInterop.Infrastructure
+{
+    internal sealed class DotNetStreamReferenceJsonConverter : JsonConverter<DotNetStreamReference>
+    {
+        private static readonly JsonEncodedText DotNetStreamRefKey = JsonEncodedText.Encode("__dotNetStream");
+
+        public DotNetStreamReferenceJsonConverter(JSRuntime jsRuntime)
+        {
+            JSRuntime = jsRuntime;
+        }
+
+        public JSRuntime JSRuntime { get; }
+
+        public override DotNetStreamReference Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            => throw new NotSupportedException($"{nameof(DotNetStreamReference)} cannot be supplied from JavaScript to .NET because the stream is released after being sent.");
```

Comment:
```suggestion
            => throw new NotSupportedException($"{nameof(DotNetStreamReference)} cannot be supplied from JavaScript to .NET because the stream contents have already been transferred.");
```